tests/vmi_multiqueue_test.go

```diff
@@ -74,10 +74,13 @@ var _ = Describe("[sig-compute]MultiQueue", func() {
 			By("Creating and starting the VMI")
 			vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
 			Expect(err).ToNot(HaveOccurred())
-			tests.WaitForSuccessfulVMIStartWithTimeout(vmi, 360)
+			vmi = tests.WaitForSuccessfulVMIStartWithTimeout(vmi, 360)
 
 			By("Checking if we can login")
 			Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+			By("Checking QueueCount has the expected value")
+			Expect(vmi.Status.Interfaces[0].QueueCount).To(Equal(uint(numCpus)))
```

Comment:
nit: Is `unit` required?
Can it be set on declaration? (line 64)

Reply:
> nit: Is `unit` required?

Required unless we change it here to uint32 which is the default type when declaring unit variable
https://github.com/kubevirt/kubevirt/blob/fcc721bb291753b85bd2aeba765743bd2ee2e7f0/staging/src/kubevirt.io/api/core/v1/types.go#L556

> Can it be set on declaration? (line 64)

I believe so, will try (unless we change the type in the link above)

Reply:
I think there is a recommendation somewhere to not use uint for API/s, because it is dependent on the architecture.
If needed, I can try and look for it.

Reply:
fwiw, there are usages of `unit` in the `types.go`

so to use uint32 ? or just int / int32 ?

Reply:
Note that the reason i used uint is that here it is uint, changing it to int or so might make the code even less cleaner

https://github.com/kubevirt/kubevirt/pull/8071/files#diff-1c532caac99cc080ba1fb740c75c0ff712ca91509093f55cebdfaefee06ae887R117

Reply:
If something exists today in a problematic form, it does not mean we should keep having that. At least that is not a good reason to do something for sure.

See [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types) for the recommendations.

Reply:
Meant to raise awareness
Thanks for the docs

Changed to int

Reply:
> Changed to int

The ref api convention document states:
> All public integer fields MUST use the Go (u)int32 or Go (u)int64 types, not (u)int (which is ambiguous depending on target platform). Internal types may use (u)int.

Per my interpretation of it, any external API, like the presented here, should not use `(u)int` as it is platform dependent.
I think we need to use `int32` in this specific case.

Reply:
It is defined as int32 in the API, which is correct. Do you mean in the test ? If so, I agree - it doesn't hurt to explicitly state the bit length of the integer.

Reply:
I think what counts is [here](https://github.com/kubevirt/kubevirt/pull/8071/commits/b0bff249d11b9451327447991e7a331bd162449b#diff-34a591e3aa47260cfad791064c20f34f98fdca4a63532fbf7184e68682f4fcb8R556). At least I think the other places are generated based on this type.
Am I wrong?

Reply:
Oh you're right! Sorry.

I had only looked at the swagger file - which is strangely correct ... My bad :crying_cat_face: